src/Joint.cc

```diff
@@ -28,33 +44,85 @@ RBDL_DLLAPI void jcalc (
   assert (joint_id > 0);
 
   if (model.mJoints[joint_id].mJointType == JointTypeRevoluteX) {
-    model.X_J[joint_id] = Xrotx (q[model.mJoints[joint_id].q_index]);
+    double s, c;
+    sincos (q[model.mJoints[joint_id].q_index], &s, &c);
+
+    model.X_lambda[joint_id].E = Matrix3d (
+        model.X_T[joint_id].E(0, 0),
+        model.X_T[joint_id].E(0, 1),
+        model.X_T[joint_id].E(0, 2),
+
+         c * model.X_T[joint_id].E(1, 0) + s * model.X_T[joint_id].E(2, 0),
+         c * model.X_T[joint_id].E(1, 1) + s * model.X_T[joint_id].E(2, 1),
+         c * model.X_T[joint_id].E(1, 2) + s * model.X_T[joint_id].E(2, 2),
+
+        -s * model.X_T[joint_id].E(1, 0) + c * model.X_T[joint_id].E(2, 0),
+        -s * model.X_T[joint_id].E(1, 1) + c * model.X_T[joint_id].E(2, 1),
+        -s * model.X_T[joint_id].E(1, 2) + c * model.X_T[joint_id].E(2, 2));
+
+    model.X_lambda[joint_id].r = model.X_T[joint_id].r;
+
     model.v_J[joint_id][0] = qdot[model.mJoints[joint_id].q_index];
   } else if (model.mJoints[joint_id].mJointType == JointTypeRevoluteY) {
-    model.X_J[joint_id] = Xroty (q[model.mJoints[joint_id].q_index]);
+    double s, c;
+    sincos (q[model.mJoints[joint_id].q_index], &s, &c);
```

Comment:
I'll need to take a closer look at this one.

Reply:
So actually casadi also depends on cmath so it overwrites sincoes itself. Anyway the solution was to rename the inline function so it can not confuse it with the buildin one.